src/components/clusterConfiguration/NetworkConfiguration.tsx

```diff
@@ -34,19 +47,67 @@ const NetworkConfiguration: React.FC<NetworkConfigurationProps> = ({ cluster, ho
     }
   };
 
+  const isMultiNodeCluster = !isSingleNodeCluster(cluster);
+  const isUserManagedNetworking = values.networkingType === 'userManaged';
+  const handleManagedNetworkingControlGroupChange = React.useCallback<
+    Required<ManagedNetworkingControlGroupProps>['onChange']
+  >(
+    (nextValue) => {
+      if (nextValue === 'userManaged') {
+        const isTouched = true;
+        const shouldValidate = true;
+        setFieldValue('vipDhcpAllocation', false);
+        setFieldValue('hostSubnet', NO_SUBNET_SET, !shouldValidate);
+        setFieldTouched('hostSubnet', !isTouched, !shouldValidate);
+        setFieldValue('ingressVip', '');
+        setFieldTouched('ingressVip', !isTouched, !shouldValidate);
+        setFieldValue('apiVip', '');
+        setFieldTouched('apiVip', !isTouched, !shouldValidate);
+      }
+    },
+    [setFieldTouched, setFieldValue],
+  );
+  const handleAvailableSubnetsControlChange = React.useCallback<
+    Required<AvailableSubnetsControlProps>['onChange']
+  >(() => {
+    if (!values.vipDhcpAllocation && !isUserManagedNetworking) {
+      validateField('ingressVip');
+      validateField('apiVip');
+    }
+  }, [isUserManagedNetworking, validateField, values.vipDhcpAllocation]);
```

Comment:
These could be better handled using useEffect when related values change. I think it would make the code a bit simpler.

Example here: https://formik.org/docs/examples/dependent-fields

Reply:
This is what I did following that example. The ref was necessary in order to remember the value in the single-node cluster scenario (what happened is that after moving to the final step and pressing back the hostSubnet is reset). In multi-node cluster the next button is disabled. At this point I think that if we can't spot what's wrong we should move forward with what we have (at least it works right now...) we can put a task aside to refactor this part later.

```typescript
const hostSubnetRef = useRef(values.hostSubnet || NO_SUBNET_SET);

  useEffect(() => {
    const isTouched = true;
    const shouldValidate = true;

    if (isUserManagedNetworking) {
      setFieldValue('vipDhcpAllocation', false);
      setFieldValue('hostSubnet', hostSubnetRef.current, !shouldValidate);
      setFieldTouched('hostSubnet', !isTouched, !shouldValidate);
      setFieldValue('ingressVip', '');
      setFieldTouched('ingressVip', !isTouched, !shouldValidate);
      setFieldValue('apiVip', '');
      setFieldTouched('apiVip', !isTouched, !shouldValidate);
    } else {
      if (!values.vipDhcpAllocation) {
        validateField('ingressVip');
        validateField('apiVip');
      }
    }
  }, [
    isUserManagedNetworking,
    setFieldTouched,
    setFieldValue,
    validateField,
    values.vipDhcpAllocation,
  ]);
```

Reply:
never mind I managed to make it work :+1: